From: George McCollister <george.mccollister@gmail.com>
Date: Wed, 19 Sep 2012 09:36:10 -0500
Subject: [PATCH] define __aligned_u64 if not defined

usr/include/sys/types.h defines_LINUX_TYPES_H which prevents the kernel
provided types.h from defining __aligned_u64. This patch just defines
__aligned_u64 in usr/include/sys/types.h.

Build against Linux 3.2.28 with gcc-4.6.2-glibc-2.14.1 fails without
this patch.

Signed-off-by: George McCollister <george.mccollister@gmail.com>
---
 usr/include/sys/types.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/usr/include/sys/types.h b/usr/include/sys/types.h
index 6780ed1..3ef6653 100644
--- a/usr/include/sys/types.h
+++ b/usr/include/sys/types.h
@@ -18,6 +18,10 @@ typedef ptrdiff_t ssize_t;
 /* Keeps linux/types.h from getting included elsewhere */
 #define _LINUX_TYPES_H
 
+#ifndef __aligned_u64
+#define __aligned_u64 __u64 __attribute__((aligned(8)))
+#endif
+
 typedef __kernel_fd_set fd_set;
 typedef uint32_t dev_t;
 typedef __kernel_ino_t ino_t;
